logtape/logger.ts

```diff
@@ -526,21 +528,41 @@ export class LoggerImpl implements Logger {
       },
       timestamp: Date.now(),
       properties: {},
-    });
+    };
+
+    let cachedProps: Record<string, unknown> | undefined = undefined;
+    Object.assign(record, typeof properties === "function" ? {
+      get properties() {
+        if (cachedProps == null) cachedProps = properties();
+        return cachedProps;
+      },
+    } : { properties });
+    this.emit(record, bypassSinks);
   }
 
   logTemplate(
     level: LogLevel,
     messageTemplate: TemplateStringsArray,
     values: unknown[],
+    properties: Record<string, unknown> | (() => Record<string, unknown>) = {},
+    bypassSinks?: Set<Sink>,
```

Comment:
I think `bypassSinks` is kind of an internal parameter, so it's probably okay not to expose it.

Reply:
Done